### ISSUE TITLE
pstree has a different callback function signature than nodemon is expecting

### ISSUE THREAD

**Issue by sincraianul** [remy/nodemon#1455]
It seems like `kids` is undefined here, because the callback for `psTree` is not error-first. Suppose this is related to the latest changes to `pstree`.

```
        spawn('kill', ['-s', sig, child.pid].concat(kids.map(p => p.PID)))
                                                         ^

TypeError: Cannot read property 'map' of undefined
```

**Comment by Rohithzr** [comment]
this issue seems to be affecting new installations on v1.18.4 too.
referencing this issue https://github.com/remy/nodemon/issues/1453

**Comment by remy** [comment]
Yeah. On it once I get to a machine (unless you can send a pr to remy/pstree to fix the signature)

**Comment by Rohithzr** [comment]
on it

**Comment by iviaks** [comment]
@sincraianul , Fixed by installing pstree.remy@1.1.0 instead of pstree.remy@1.1.1

**Comment by remy** [comment]
1.1.2 is up and has the fix. If you uninstall and re-install it should work.

I'm pushing a version of nodemon that requires that version too.

**Comment by Rohithzr** [comment]
@pavanSaberjack
well technically you shouldn't see this issue, try uninstalling nodemon and make sure local version is missing.
then install it again, if it still doesnt work, try installing remy 1.1.2 manually

**Comment by sysrun** [comment]
Same - looks like the process is not killed on restart

**Comment by RoyalHunt** [comment]
Same for us...  1.1.2 fixed one problem, but brought another, with not killed process on update.

**Comment by remy** [comment]
I'm going to (either) open a new issue for this or discuss on an open thread. I couldn't replicate this locally, but I *can* inside of a travis/linux like environment…

**Comment by Nax** [comment]
Can replicate the `EADDRINUSE` issue.

 * macOS 10.13.6
 * node 11.2.0
 * nodemon 1.18.6
 * pstree.remy 1.1.2

Everything is fine if nodemon is used to directly start a script, but if using `babel-node`, my code never receive `SIGUSR2`. Looks like it kills `babel-node` but not it's child.

**Comment by Rohithzr** [comment]
@iviaks is this resolved?

**Comment by pavanSaberjack** [comment]
@Rohithzr  For alternate method, I tried installing nodemon v 1.18.3 which just works fine.
